src/libraries/Microsoft.PowerFx.Core/Public/Config/DeferredSymbolTable.cs

```diff
@@ -37,6 +37,7 @@ public DeferredSymbolTable(DisplayNameProvider map, Func<string, string, Formula
         }
 
         // SymbolTable is conceptually constant. 
+        [ThreadSafeProtectedByLock("Interlocked.Increment")]
```

Comment:
>[ThreadSafeProtectedByLock("Interlocked.Increment")] [](http://example.com/codeflow?start=8&length=52)

what does this mean?

It looks like it's not protected by a lock; but instead, it's just never modified. 
But VersionHash itself is mutable, so that' it's readonly is not enough protection. 

this may need more discussion.

Reply:
interlocked.increment is thread safe as in https://learn.microsoft.com/en-us/dotnet/api/system.threading.interlocked.increment?redirectedfrom=MSDN&view=net-7.0#System_Threading_Interlocked_Increment_System_Int32__
(built in lock in threading, not our own lock).
So if we make this change, VersionHash should be immutable. 
Should I create a different thread safe tag for this case? Or you still think VersionHash is not thread safe?